diff --git a/node_modules/@gonetone/get-youtube-id-by-url/src/GetYouTubeIdByUrl.js b/node_modules/@gonetone/get-youtube-id-by-url/src/GetYouTubeIdByUrl.js
index cdf15a1..89b266d 100644
--- a/node_modules/@gonetone/get-youtube-id-by-url/src/GetYouTubeIdByUrl.js
+++ b/node_modules/@gonetone/get-youtube-id-by-url/src/GetYouTubeIdByUrl.js
@@ -61,8 +61,15 @@ const checkUrl = (url) => url.indexOf('youtube.com') !== -1 || url.indexOf('yout
 const channelId = async (url) => {
   if (checkUrl(url)) {
     const ytChannelPageResponse = await axiosInstance.get(url)
-    const $ = cheerio.load(ytChannelPageResponse.data)
+    const searchChannelIdIndex = ytChannelPageResponse.data.search('channelId')
+    if (searchChannelIdIndex !== -1) {
+      const channelId = ytChannelPageResponse.data.slice(searchChannelIdIndex + 12, searchChannelIdIndex + 13 + 23)
+      if (channelId) {
+        return channelId
+      }
+    }
 
+    const $ = cheerio.load(ytChannelPageResponse.data)
     const id = $('meta[itemprop="channelId"]').attr('content')
     if (id) {
       return id
