diff --git a/node_modules/canvacord/scripts/postinstall.mjs b/node_modules/canvacord/scripts/postinstall.mjs
index 62cd3a7..9a5ef4b 100644
--- a/node_modules/canvacord/scripts/postinstall.mjs
+++ b/node_modules/canvacord/scripts/postinstall.mjs
@@ -1,4 +1,4 @@
 import { build } from "./assets/build.mjs"
 
 // make sure to build assets if executed through postinstall script
-await build(true);
\ No newline at end of file
+build(true);
\ No newline at end of file
